lib/queuePopulator/LogReader.js

```diff
@@ -330,15 +318,29 @@ class LogReader {
             }
             return key;
         }
-        if (entry.key === undefined || entry.value === undefined) {
-            return;
+        // TODO: handle this better, this is too specific to notifications
+        // temp fix
+        const notifExtension = this._extensions.find(ext => (
+            ext instanceof NotificationQueuePopulator
+        ));
+        if (notifExtension) {
+            notifExtension.filter({
```

Comment:
This section looks outdated now, you should merge the latest changes there.

Reply:
This is the base pr for the other pr. I was thinking about merging this first.

Reply:
I see no point in merging a commit that we know is going to be changed immediately, it makes the history more confusing than it should be IMHO.

Reply:
The commit now has working code from #1006. Not the latest changes from #1006